tests/integration/targets/iso_customize/tasks/iso_customize.yml

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2022, Ansible Project
+# Copyright (c) 2022, VMware, Inc. All Rights Reserved.
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Add a line to the file test02.cfg and make sure it succeed
+  ansible.builtin.lineinfile:
+    path: "{{ test_dir }}/test02.cfg"
+    regexp: "^test"
+    line: "test"
+
+- name: "Customize ISO file"
+  community.general.iso_customize:
+    src_iso: "{{ test_dir }}/test.iso"
+    dest_iso: "{{ test_dir }}/{{ dest_iso_name }}"
+    delete_files:
+      - "/test01.cfg"
+      - "/test02.cfg"
+    add_files:
+      - src_file: "{{ test_dir }}/test01.cfg"
+        dest_file: "/preseed/ubuntu.seed"
+      - src_file: "{{ test_dir }}/test02.cfg"
+        dest_file: "/test02.cfg"
+
+- debug: var=ansible_distribution
+
+# Only need to check the files is deleted/added as the requested modifications 
+# in one type of OS(take MacOS here)
+- block:
+    - name: "Mount customized ISO on MAC"
+      command: "hdiutil attach {{ test_dir }}/{{ dest_iso_name }} -mountroot {{ test_dir }}/iso_mount"
+
+    # For MAC, we have different root directory for different type of ISO
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/disk_image"
+
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/AUTOINSTALL"
+      when: dest_iso_name.find('joliet') != -1
+
+    - set_fact:
+        mount_root_dir: "{{ test_dir }}/iso_mount/CDROM"
+      when: dest_iso_name.find('udf') != -1
+
+    - debug: var=mount_root_dir
+
+    - name: Check the file test01.cfg is deleted
+      stat:
+        path: "{{ mount_root_dir }}/test01.cfg"
+      register: check_file
+
+    - assert:
+        that:
+        - check_file.stat.exists == False
+
+    - name: Check the file /preseed/ubuntu.seed is added
+      stat:
+        path: "{{ mount_root_dir }}/preseed/ubuntu.seed"
+      register: check_file
+
+    - assert:
+        that:
+        - check_file.stat.exists == True
+
+    - name: Get the content of file test02.cfg
+      set_fact:
+        file_contents: "{{ lookup('file', mount_root_dir + '/test02.cfg') }}"
+
+    - fail: msg="Failed to replace the file test02.cfg"
+      when: file_contents != "test"
+
+    - name: Umount ISO
+      mount:
+        path: "{{ mount_root_dir }}"
+        fstab: "{{ test_dir }}/temp.fstab"
+        state: unmounted
+  when: ansible_distribution == "MacOSX"
```

Comment:
Only noticed now: is the test running only on Mac?? I don't see anything else happening when `ansible_distribution != "MacOSX"`.

Reply:
> Only noticed now: is the test running only on Mac?? I don't see anything else happening when `ansible_distribution != "MacOSX"`.

 I run the main.yml in both MAC and Ubuntu.
 For checking changed files inside ISO, it will be enough to test in only one type of OS(take Mac here). If it pass, we can know iso_customized.py works well.

Thanks
Yuhua Zou

Reply:
Running in Ubuntu when the test does not check anything in Ubuntu serves no purpose. Please expand the testing:
* to test in Linux as well (ideally in others systems, but we can add those tests later on)
* to include more meaningful use cases, such as:
  * no add_files
  * no delete_files
  * none of both
  * deleting non-existing files
  * add files with invalid source path
  * filenames with whitespaces and also with other UTF-8 characters, like á, ã, Mandarin symbols, etc..

Reply:
I add more testcases for it. Thanks

Except the following one. The other testcases you mentioned are added
 > filenames with whitespaces and also with other UTF-8 characters, like á, ã, Mandarin symbols, etc..

I am considering to add it. Thanks.

Reply:
add more testcase:
    no add_files
    no delete_files
    none of both
    deleting non-existing files
    add files with invalid source path
    filenames with whitespaces
    invalid dest directory